modules/client-common/test/schemas.test.ts

```diff
@@ -0,0 +1,951 @@
+      const uint8Array: number[] = [];
+      expect(() => Uint8ArraySchema.strict().validateSync(uint8Array)).toThrow(
+        new ValidationError(new InvalidParameter("Uint8Array").message),
+      );
+    });
+  });
+
+  describe("Test IpfsCid", () => {
+    it("should validate a valid v0 ipfs CID", () => {
+      const ipfsUri = TEST_IPFS_URI_V0;
+      IpfsUriSchema.strict().validateSync(ipfsUri);
+    });
+    it("should validate a valid v1 ipfs CID", () => {
+      const ipfsUri = TEST_IPFS_URI_V1;
+      IpfsUriSchema.strict().validateSync(ipfsUri);
+    });
+    it("should throw an error if the ipfs CID is invalid", () => {
+      const ipfsUri = TEST_INVALID_IPFS_URI;
+      expect(() => IpfsUriSchema.strict().validateSync(ipfsUri)).toThrow(
+        new ValidationError(new InvalidCidError().message),
+      );
+    });
+  });
+  describe("Test Subdomain", () => {
+    it("should validate a valid subdomain", () => {
+      const subdomain = TEST_SUBDOMAIN;
+      SubdomainSchema.strict().validateSync(subdomain);
+    });
+    it("should throw an error if the subdomain is invalid", () => {
+      const subdomain = TEST_INVALID_SUBDOMAIN;
+      expect(() => SubdomainSchema.strict().validateSync(subdomain)).toThrow(
+        new ValidationError(new InvalidSubdomainError().message),
+      );
+    });
+  });
+
+  describe("Test Pagination", () => {
+    it("should validate a valid Pagination", () => {
+      const pagination: Pagination = {
+        skip: 0,
+        limit: 1,
+        direction: SortDirection.ASC,
+      };
+      PaginationSchema.strict().validateSync(pagination);
+    });
+    it("should validate a valid Pagination without optional params", () => {
+      const pagination: Pagination = {};
+      PaginationSchema.strict().validateSync(pagination);
+    });
+    it("should throw an error if the skip is invalid", () => {
+      const pagination: Pagination = {
+        skip: -1,
+        limit: 1,
+      };
+      expect(() => PaginationSchema.strict().validateSync(pagination)).toThrow(
+        new ValidationError(
+          "skip must be greater than or equal to 0",
+        ),
+      );
+    });
+    it("should throw an error if the limit is invalid", () => {
+      const pagination: Pagination = {
+        skip: 0,
+        limit: 0,
+      };
+      expect(() => PaginationSchema.strict().validateSync(pagination)).toThrow(
+        new ValidationError(
+          "limit must be greater than or equal to 1",
+        ),
+      );
+    });
+    it("should throw an error if the direction is invalid", () => {
+      const pagination = {
+        skip: 0,
+        limit: 1,
+        direction: "invalid",
+      };
+      expect(() => PaginationSchema.strict().validateSync(pagination)).toThrow(
+        new ValidationError(
+          "direction must be one of the following values: asc, desc",
+        ),
+      );
+    });
+  });
+
+  describe("Test MultiTargetPermission", () => {
```

Comment:
The contracts throw if the `who` and the `where` are set to then ANY_ADDR (`0xffffffffffffffffffffffffffffffffffffffff`). Idk if you wanna check this too in your validation